### PR TITLE
GatewayController remove REST calls from service

### PR DESCRIPTION
**Pull request by RadoslavMonov**

### CODE REVIEW

File: src/main/java/com/monov/student/controller/StudentController.java [hunk 1]
```diff
@@ -35,17 +33,15 @@ public Student findStudentById(@PathVariable("id") Long studentId) {
...
-    }
-
     @PostMapping("/{studentId}/{courseId}")
     public Student addCourseToStudent(@PathVariable("studentId") Long studentId,
                                       @PathVariable("courseId") Long courseId) {
         log.info("Inside addCourseToStudent method of StudentController ");
         return  studentService.addCourseToStudent(studentId,courseId);
     }
 
+    @GetMapping("/course/{id}")
```
Comment (SavageStan) [inline src/main/java/com/monov/student/controller/StudentController.java]: Collection names in URI's should be plural.

File: src/main/java/com/monov/student/repository/StudentRepository.java [hunk 2]
```diff
@@ -2,10 +2,11 @@
 
 import com.monov.student.entity.Student;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface StudentRepository extends JpaRepository<Student, Long> {
+public interface StudentRepository extends JpaRepository<Student, Long>, JpaSpecificationExecutor<Student> {
```
Comment (SavageStan) [inline src/main/java/com/monov/student/repository/StudentRepository.java]: Correct me if I am wroing, but this JpaSpecificationExecutor interface seems to be related to CriteriaBuilder API? Are we using it?

File: src/main/java/com/monov/student/service/StudentService.java [hunk 3]
```diff
@@ -40,20 +32,16 @@ public Student findStudentById(Long studentId) {
...
-    }
-
     public Student addCourseToStudent(Long studentId, Long courseId) {
         Student studentToEdit = findStudentById(studentId);
         studentToEdit.getCourseIds().add(courseId);
         studentRepository.save(studentToEdit);
         return  studentToEdit;
     }
 
+    // To Do
```
Comment (SavageStan) [inline src/main/java/com/monov/student/service/StudentService.java:42]: Just a heads up, in order to correctly use TODO comments to be useful, they need to include the date of their creation, the name of the person that created them and a link to a Jira story that should describe the work that is to be done.

Comment (RadoslavMonov) [reply src/main/java/com/monov/student/service/StudentService.java:42]: Done

File: src/main/java/com/monov/student/controller/StudentController.java (hunk 1 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/student/controller/StudentController.java]: done

File: src/main/java/com/monov/student/repository/StudentRepository.java (hunk 2 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/student/repository/StudentRepository.java]: Removed. Done

**Review by SavageStan — APPROVED** [review]

**Pull request merged**